tracer/test/snapshots/AspNetMvc4Tests.Classic.WithFF.__path=_Home_BadRequestWithStatusCode-statuscode=401&TransferRequest=true_statusCode=500.verified.txt

```diff
@@ -14,11 +14,15 @@
 System.Exception: Oops, it broke. Specified status code was: 401
 at Samples.AspNetMvc4.Controllers.HomeController.BadRequestWithStatusCode(Int32 statuscode),
       error.type: System.Exception,
+      http.client_ip: 127.0.0.1,
       http.method: GET,
       http.request.headers.host: localhost:00000,
+      http.route: /home/badrequestwithstatuscode,
```

Comment:
I was expecting the route to be `{controller}/{action}/{id}` since the child span had the route information in the `aspnet.route` tag 🤔

Reply:
Interestingly, this looks like an issue with the `WithFF` configuration, because the Integrated version of this same request also shows this issue. I'll have to dig deeper 🕵🏼

Reply:
I think I understand what's going on. I think we're entering `AspNetResourceNameHelper.CalculateResourceName` and we're setting `http.route` to be equivalent to the `Resource`, just without the HTTP method. This looks to happen with both the `WithFF` scenario and the `WithExpansion` scenario.

Let's use this comment thread to figure out if we should be doing purely the route template or trying to fill it in. I was thinking that the variables we have to expand routes would only apply to the resource names and the `http.route` would be the most general value possible. I'm wondering what others think here

Reply:
so the [RFC ](https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#http.route)seems to say that `area, controller `and `action `should be resolved but not `{id} `considered as a route parameter. But somehow, controller and actions could also be considered parameters, although, I think in the mvc cycle at some point, only {id} appears in the route params and controller and action are removed. I thought  the env variable `DD_TRACE_EXPAND_ROUTE_TEMPLATES_ENABLED` made sense also for `http.route`, but it's true that it depends on the flag `DD_TRACE_ROUTE_TEMPLATE_RESOURCE_NAMES_ENABLED`  which, according to its names, should apply only to resources, so maybe `http.route` should only depend on the first one... 

As the tag is optional and at best effort, I thought it didnt matter too much if it was varying a little.. but I can remove it alltogether for this PR and we could add it later once we know more 😄

Reply:
So in the end, I ll just put the same content as in the aspnet_route and aspnetcore_route tags like mentioned in [this comment](https://github.com/DataDog/dd-trace-dotnet/pull/2915#discussion_r920438053)

Reply:
> so the [RFC ](https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#http.route)seems to say that area, controller and action should be resolved but not {id}

I feel like we're looking at different documents, the RFC seems to say basically nothing other than "don't expand parameters" 😅 

> But somehow, controller and actions could also be considered parameters, although, I think in the mvc cycle at some point, only {id} appears in the route params and controller and action are removed

Only by _us_ as part of ResourceName, they're still parameters in the strict sense..

> So in the end, I'll just put the same content as in the aspnet_route and aspnetcore_route tags like mentioned

I think that makes sense, thanks 🙂